gax-grpc/src/main/java/com/google/api/gax/grpc/GrpcCallableFactory.java

```diff
@@ -0,0 +1,328 @@
+            batchingCallSettings.getBatchingSettings(),
+            clientContext.getExecutor(),
+            batchingCallSettings.getFlowController());
+    callable =
+        new BatchingCallable<>(
+            callable, batchingCallSettings.getBatchingDescriptor(), batcherFactory);
+    callable =
+        new EntryPointUnaryCallable<>(
+            callable, GrpcCallContext.createDefault(), getCallContextEnhancers(clientContext));
+    return new BatchingCreateResult<>(batcherFactory, callable);
+  }
+
+  /**
+   * Creates a callable object that represents a long-running operation. Designed for use by
+   * generated code.
+   *
+   * @param directCallable the callable that directly issues the call to the underlying API
+   * @param operationCallSettings {@link OperationCallSettings} to configure the method-level
+   *     settings with.
+   * @param clientContext {@link ClientContext} to use to connect to the service.
+   * @param operationsStub {@link OperationsClient} to use to poll for updates on the Operation.
+   * @return {@link com.google.api.gax.rpc.OperationCallable} callable object.
+   */
+  public static <RequestT, ResponseT extends Message, MetadataT extends Message>
+      OperationCallable<RequestT, ResponseT, MetadataT, Operation> create(
+          UnaryCallable<RequestT, Operation> directCallable,
+          OperationCallSettings<RequestT, ResponseT, MetadataT, Operation> operationCallSettings,
+          ClientContext clientContext,
+          OperationsStub operationsStub) {
+    GrpcOperationCallableImpl<RequestT, ResponseT, MetadataT> callableImpl =
+        createImpl(directCallable, operationCallSettings, clientContext, operationsStub);
+    return new EntryPointOperationCallable<>(
+        callableImpl, GrpcCallContext.createDefault(), getCallContextEnhancers(clientContext));
+  }
+
+  static <RequestT, ResponseT extends Message, MetadataT extends Message>
+      GrpcOperationCallableImpl<RequestT, ResponseT, MetadataT> createImpl(
+          UnaryCallable<RequestT, Operation> directCallable,
+          OperationCallSettings<RequestT, ResponseT, MetadataT, Operation> operationCallSettings,
+          ClientContext clientContext,
+          OperationsStub operationsStub) {
+
+    UnaryCallable<RequestT, Operation> initialCallable =
+        GrpcCallableFactory.createBaseCallable(
+            directCallable, operationCallSettings.getInitialCallSettings(), clientContext);
+
+    RetryAlgorithm<Operation> pollingAlgorithm =
+        new RetryAlgorithm<>(
+            new OperationResponsePollAlgorithm(), operationCallSettings.getPollingAlgorithm());
+    ScheduledRetryingExecutor<Operation> scheduler =
+        new ScheduledRetryingExecutor<>(pollingAlgorithm, clientContext.getExecutor());
+
+    return new GrpcOperationCallableImpl<>(
+        initialCallable,
+        clientContext,
+        scheduler,
+        operationsStub,
+        new GrpcOperationCallableImpl.ResponseTransformer<>(
```

Comment:
I will instead pass `operationCallSettings` to the `GrpcOperationCallableImpl` constructor, which can create the transformers itself. That simplifies the code I think.